content/guides/pedestal-with-component.adoc

```diff
@@ -1,16 +1,419 @@
+If you get stuck at any point in this guide, please submit an
+https://github.com/pedestal/docs/issues[issue] about this guide or hop
+over to the
+https://groups.google.com/forum/#!forum/pedestal-users[mailing list]
+and raise your hand there.
+
+This guide shows fragments of code as we add them. Sometimes it helps
+to see the whole thing at once, so you can always check out
+<<_the_whole_shebang,the whole shebang>> at the end of this guide.
+
+== Where We Are Going
+In this guide, we're going to step through creating a Pedestal service
+using Component. We'll start off by creating a Pedestal component and
+wire it into a Component system map. We'll then proceed to testing our
+service.
+
+== Before We Begin
+What responsibilities do we want the Pedestal component to have? For
+the purposes of this guide, it should only be responsible for the
+initialization of the Pedestal Provider and starting of the
+server. It's configuration, the Pedestal service map, should be a
+component dependency and should not be managed by the Pedestal
+component.
+
+The Pedestal component should also be amiable to testing through
+Pedestal's testing facilities. This should only require a Pedestal
+service-fn to be available.
+
+== A Simple Pedestal Component
+
+Let's start with the Pedestal component. Create a `src` directory with
+a `pedestal.clj` file.
+
+[source, clojure]
+.src/pedestal.clj
+
+----
+include::component/src/pedestal.clj[tags=ns]
+----
+<1> Create a _pedestal_ namespace to house the Pedestal component.
+<2> We need to require `com.stuartsierra.component` to implement the
+`start` and `stop` `Lifecycle` methods.
+<3> We need to require the `io.pedestal.http` for server creation,
+starting and stopping.
+
+Let's start implementing the component.
+----
+include::component/src/pedestal.clj[tags=component-init]
+----
+<1> Create a _Pedestal_ record. This record will contain a number of
+ fields, the first field is _service-map_ which will hold the Pedestal
+ service map component dependency.
+<2> The next field is _start-fn_, this will be passed in through the
+ component constructor and will be used to start the Pedestal
+ server. This function must accept one parameter, the Pedestal service
+ map, and return a Pedestal service map.
+<3> The next field is _stop-fn_, like _start-fn_, the stop function
+ will be passed in through the component constructor. It's responsible
+ for stopping the Pedestal server. This function must accept one
+ parameter, the Pedestal service map, and return a Pedestal service map.
+<4> The _service_ field is created as part of the component
+ initialization process.
+<2> Include the _component/Lifecycle_ protocol since we'll be
+implementing it's methods next.
+
+We'll first implement the `start` method.
+----
+include::component/src/pedestal.clj[tags=component-start]
+----
+<1> The _start_ method contains our component initialization code.
+<2> Our component's initialization process will be idempotent. If
```

Comment:
Words like "idempotent" are common in the FP space, but unfamiliar to many programmers. It would be friendlier to expand the idea first, then introduce the word (perhaps in a parenthetical?)